### PR TITLE
fixup CI for bee image release

### PR DESCRIPTION
**Pull request by lgadban**
removed the dockerx and qemu steps since we're currently only building for amd64

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         uses: styfle/cancel-workflow-action@0.4.0
         with:
           access_token: ${{ github.token }}
-      - name: Check out code into the Go module directory
+      - name: Check repository
```
Comment (EItanya) [inline .github/workflows/release.yaml:20]: nit: 
```suggestion
      - name: Checkout repository
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -74,13 +74,16 @@ jobs:
...
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@master
+      - name: Set up Go 1.17
+        uses: actions/setup-go@v1
         with:
-          platforms: all
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@master
+          go-version: 1.17.3
```
Comment (EItanya) [inline .github/workflows/release.yaml:80]: nit: 1.17.6 is latest
```suggestion
          go-version: 1.17.6
```

File: .github/workflows/release.yaml (hunk 1 above)
Comment (lgadban) [reply .github/workflows/release.yaml]: 🤦‍♂️

File: .github/workflows/release.yaml (hunk 2 above)
Comment (lgadban) [reply .github/workflows/release.yaml]: updated all steps to use 1.17.6; we can parameterize this as the release grows in complexity?

**Review by EItanya — APPROVED** [review]

**Pull request merged**